src/core/solver.py

```
@@ -272,69 +298,253 @@ def add_scoring_objectives(self, strategy: ScoringStrategy) -> None:
+        # Total participants = num_people
+        max_tie_breaker = (
+            (self.num_groups - 1) * (self.num_people - 1) * self.num_people
+        )
+        total_bound = sum(self.objective_bounds) + max_tie_breaker
+
+        if total_bound > max_int_limit:
+            raise ValueError(
+                f"Aggregate objective theoretical maximum ({total_bound}) "
+                f"exceeds CP-SAT safety bound ({max_int_limit}). "
+                "Consider reducing score weights or group count."
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)